docs/structured-streaming-state-data-source.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: global
+displayTitle: State data source (Experimental) Guide in Structured Streaming
+title: State data source (Experimental) Guide in Structured Streaming
```

Comment:
This is the name of doc link in programming guide. I think it is better to keep them same.

```suggestion
title: State Data Source Integration Guide
```